### ISSUE TITLE
Sonos speakers not showing on iPhone with Airsonos on windows

### ISSUE THREAD

**Issue by Shad-Shad** [stephen/airsonos#390]
Hi,
I have installed airsonos on Raspberry pi and it worked great (thank you). the only issue I had was the audio quality. I understand this is because the Pi does not have enough power to handle.

So, i decided to install this on another computer that i have that is running Windows 10. the instillation was successful and when I run airsonos in the command prompt, it finds both my speakers. but that's it: The speakers don't appear on the iPhone Airplay devices.

I have a Denon Hi-fi system that is AirPlay compatible and it appears on the iPhone and i can send the music to it with no issues. I also have an HP printer and i can print to it from my iPhone through AirPrint.

So, not sure what the issue is if everything works fine when running on Pi (except for the audio quality) and everything else that is Airplay works with no issues. 

Thanks for your help

**Comment by Tiger519** [comment]
@Shad-Shad TL;DR the Windows build doesn't work and needs to be updated.

From ios9 forward there was a change that caused Airsonos sources not to show up. This was fixed in the main build, but the Windows build was never updated. I pinged @stephen in a separate issue on this, but have not heard back. There haven't been any updates to this project in a while so I'm not sure he is still working on it. If someone knows how to create a Windows build like was done previously, I'd be willing to try. I have an RPi Zero but haven't bothered with Airsonos on it because it's my Homebridge server and I don't want to risk bringing it down (plus I think the Zero won't have good enough performance for it).

**Comment by Shad-Shad** [comment]
Thanks @Tiger519. Pity this is not being developed anymore.
I ran on the RPi 3 but the sound is very choppy. I don't think it's a performance issue since the CPU usage was at 15%.

maybe there were further changes in iOS and Sonos that have not been included.

Anyways, AirPlay2 should be out soon and Sonos is planning to support it. Let's wait and see

**Comment by Tiger519** [comment]
@Shad-Shad I ran across a similar application made (and being maintained) by @philippe44, [AirConnect](https://github.com/philippe44/AirConnect). You should be able to run that on your Pi3 just fine. The README is really good and has a few tweaks you need to make to run on Sonos. The only drawback is that you have to introduce a buffering delay for it to work reliably, so you can't use multi-room through iTunes as you're going to have about a 3 second delay between sources.

FYI Sonos has not committed to anything specifically on older devices. We just know that the Sonos ONE, and depending on which message you believe, Play:5 gen 2 and Playbase, will support it. It is unknown if older speakers will get it and/or if having one of the newer devices in your ecosystem will enable all as Airplay destinations.

**Comment by Shad-Shad** [comment]
@Tiger519 thank you so much. I tried AirConnect and it works great. I was not thinking of multi-room since i doubt AirPlay supports it. This should come with AirPlay2.

**Comment by Tiger519** [comment]
@Shad-Shad FWIW AirPlay does support multi-room, but only through iTunes. I have sent audio to three devices (two AppleTVs and an Airplay speaker) with little noticeable lag between rooms. For those looking to use AirConnect in this situation, it won't work because of the lag introduced to make it work. We will have to wait for official Sonos Airplay 2 support for that.

**Comment by philippe44** [comment]
As long as you use Sonos, you can have multi-room, you just need to create the groups through the Sonos app. The slave players will disapear and only the master will stay in the list of AirPlay devices.

**Comment by Tiger519** [comment]
I should clarify my statement. If you are trying to mix Airplay destinations through AirConnect and native Airplay, it won't work. If you have all Sonos, this would work fine, and actually I think native Airplay devices should show up in AirConnect too (e.g. AppleTV? I don't know, I have it set only for the Sonos), so if you run everything through AirConnect it should be fine. I was worried about performance so I just use AirConnect for the Sonos when I'm streaming just to my office, and use native Airplay when I'm doing multi-room (with Sonos thorugh my AppleTV, which unfortunately supports only ambient music playback through Sonos).

**Comment by philippe44** [comment]
You're absolutely right - Sonos is very protective of the way they do their synchronization, so no option to do anything there
